bitcoin/src/blockdata/transaction.rs

```diff
@@ -131,6 +144,7 @@ impl OutPoint {
     /// assert!(tx.input[0].previous_output.is_null());
     /// ```
     #[inline]
+    #[deprecated(since = "TBD", note = "use outpoint == OutPoint::COINBASE_PREVOUT instead")]
```

Comment:
We could have `is_coinbase_prevout` though I'm not sure if it's even useful.

Reply:
We'd use it internally probably, since it's a little shorter to type than the equality thing. But agreed, it doesn't seem very useful other than as a way to save a bit of typing. (And in general this isn't a super common check to do.)

Reply:
The thing is, unless our CI is broken, we already aren't using `is_null`. There's no line modifying `is_null()` calls in this PR and no `#[allow(deprecated)]` and the linter doesn't complain about use of a deprecated item.

Reply:
True -- and I don't recall ever reviewing any code using `is_null`.

So that's good evidence that it's not useful, if we aren't even using it intenally. External users have even less reason to inspect a prevout to see if it's the coinbase.

I think in practice you always know -- either you explicitly construct a coinbase input or you don't.

Reply:
Yeah I added and remove that function during dev after coming to the conclusions it was unnecessary.